### PR TITLE
Provides visualization for the simulator.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

## Summary

 - Proposes a visualization api for allow further visualizations
 - Provides an implementation using `plotly`.
 - Provides tool for getting an obj out of maliput road network
 - Provides an example

## Test it
After colcon build and source install folder:
```
python3 src/maliput_sim/examples/rail_car_agent_viz.py
```
image

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [x] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

**Review by francocipollone — COMMENTED** [review]
Some comments on the implementation

File: examples/rail_car_agent_viz.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+class LaneDirection(Enum):
+    FORWARD="0"
+    BACKWARD="1"
+
+def get_next_lane_end(lane, lane_direction):
+    default_branch = lane.GetDefaultBranch(Which.kFinish if lane_direction == LaneDirection.FORWARD else Which.kStart)
+    return default_branch
+
+class RailCarController:
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply examples/rail_car_agent_viz.py]: Controller related stuff was moved down to PR https://github.com/maliput/maliput_sim/pull/11

File: examples/rail_car_agent_viz.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    sim.add_agent(initial_agent_state_2, rail_car_2.rail_car_controller)
+
+    initial_agent_state_3 = AgentInitialState(
+        "agent_3", [240., -60., 0.], [0., 0., 0., 1], 30., 0.)
+    rail_car_3 = RailCarController()
+    sim.add_agent(initial_agent_state_3, rail_car_3.rail_car_controller)
+
+    # Visualization
+    viz = PlotlyVisualization(sim)
+    sim.add_step_hook(viz.update)
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply examples/rail_car_agent_viz.py]: As no live visualization is necessary this api has changed.

File: examples/rail_car_agent_viz.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+    # Run Simulation for 100 seconds
+    print("running...")
+    sim.step_for(100)
+    print("Sim time: ", sim.get_sim_state().sim_time)
+    viz.display()
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply examples/rail_car_agent_viz.py]: As no live visualization is necessary this api has changed.

File: examples/rail_car_agent_viz.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+from maliput_sim.viz.visualization import PlotlyVisualization
+
+import maliput.plugin
+import maliput.utility
+
+from maliput.api import LanePosition, InertialPosition, Which
+
+from enum import Enum
+
+class LaneDirection(Enum):
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply examples/rail_car_agent_viz.py]: Indeed

**Comment by francocipollone** [comment]
This PR has been split into several.
This PR goes on top of #11  and focuses on providing a visualization using plotly.

File: maliput_sim/viz/base.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from abc import ABC, abstractmethod
+
+from maliput_sim.simulation import Simulation
+
+
+class Visualization(ABC):
```
Comment (agalbachicar) [inline maliput_sim/viz/base.py:35]:
```suggestion
class Visualizer(ABC):
```

File: maliput_sim/viz/plotly.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+        # Add traces
+        for agent_name, poses in self._agent_poses.items():
+            scatter = go.Scatter3d(
+                x=[p[0] for p in poses],
+                y=[p[1] for p in poses],
+                z=[p[2] for p in poses],
+                name=agent_name,
+                mode='lines',
+                line=dict(
+                    width=5
```
Comment (agalbachicar) [inline maliput_sim/viz/plotly.py:111]: Let's move the width to a class constant.

File: maliput_sim/viz/plotly.py [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+        # Create a Mesh3d trace
+        mesh = go.Mesh3d(
+            x=[v[0] for v in vertices],
+            y=[v[1] for v in vertices],
+            z=[v[2] for v in vertices],
+            i=[f[0] for f in faces],
+            j=[f[1] for f in faces],
+            k=[f[2] for f in faces],
+            color='lightblue',  # Customize the color if desired
```
Comment (agalbachicar) [inline maliput_sim/viz/plotly.py:77]: Let's move the lightblue to a class constant.

**Review by agalbachicar — CHANGES_REQUESTED** [review]
Just a few nits.

File: maliput_sim/viz/plotly.py (hunk 7 above)
Comment (francocipollone) [reply maliput_sim/viz/plotly.py:77]: Done.

File: maliput_sim/viz/plotly.py (hunk 6 above)
Comment (francocipollone) [reply maliput_sim/viz/plotly.py]: Done.

File: maliput_sim/viz/base.py (hunk 5 above)
Comment (francocipollone) [reply maliput_sim/viz/base.py]: Done.

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**